### PR TITLE
[docs] - apidoc for AssetSelection and AssetsDefinition[CON-331]

### LINKED ISSUE #8487: include AssetSelection and AssetsDefinition in api doc

### PR DESCRIPTION
**Pull request by sryza**
Fixes #8487 

### How I Tested These Changes

Manual inspection of local docs site.

### CODE REVIEW

**Comment by sryza** [comment]
@erinkcochran87 ping for another review on this. Hoping to get this in for the release this week.

File: python_modules/dagster/dagster/core/asset_defs/asset_selection.py [hunk 1]
```diff
@@ -16,29 +16,72 @@
...
+    You can use the "|" and "&" operators to create unions and intersections of asset selections,
+    respectively.
+
+    AssetSelections are typically used with :py:func:`define_asset_job`.
+
+    Examples:
+
+        .. code-block:: python
+
+            # Select all assets in group "marketing":
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/core/asset_defs/asset_selection.py:31]: this should just be AssetSelection.groups("marketing")

**Review by erinkcochran87 — APPROVED** [review]
lgtm 🔥

**Review by OwenKephart — APPROVED** [review]
had one comment, otherwise this looks good

**Pull request merged**